SmartDeviceLink/public/SDLFileManager.m

```diff
@@ -178,11 +178,16 @@ - (void)didEnterStateFetchingInitialList {
 
         // If there was an error, we'll pass the error to the startup handler and cancel out
         if (error != nil) {
-            // HAX: In the case we are DISALLOWED we still want to transition to a ready state. Some head units return DISALLOWED for this RPC but otherwise work.
-            if([error.userInfo[@"resultCode"] isEqualToEnum:SDLResultDisallowed]) {
+            if ([error.userInfo[@"resultCode"] isEqualToEnum:SDLResultDisallowed]) {
+                // HAX: In the case we are DISALLOWED we still want to transition to a ready state. Some head units return DISALLOWED for this RPC but otherwise work.
                 SDLLogW(@"ListFiles is disallowed. Certain file manager APIs may not work properly.");
                 [weakSelf.stateMachine transitionToState:SDLFileManagerStateReady];
                 BLOCK_RETURN;
+            } else if ([error.userInfo[@"resultCode"] isEqualToEnum:SDLResultEncryptionNeeded]) {
+                // HAX: If the module rejects the ListFiles request because it requires that the request be encrypted, we still want to transition to a ready state. Unfortunately, since we do not know what files are on the module already, we may end up doing unnecessary duplicate file uploads.
```

Comment:
```suggestion
                // If the module rejects the ListFiles request because it requires that the request be encrypted, we still want to transition to a ready state. Unfortunately, since we do not know what files are on the module already, we may end up doing unnecessary duplicate file uploads.
```

This isn't a HAX because it's not working around a module bug.